### PR TITLE
Auh callback

### PR DESCRIPTION
**Pull request by erudenko**

### CODE REVIEW

File: jwt/service/jwt_token_service.go [hunk 1]
```diff
@@ -170,13 +170,18 @@ func (ts *JWTokenService) NewAccessToken(u model.User, scopes []string, app mode
...
 
-	payload := make(map[string]string)
+	payload := make(map[string]interface{})
 	if contains(app.TokenPayload(), PayloadName) {
 		payload[PayloadName] = u.Username()
 	}
 	if requireTFA {
 		payload[PayloadTFAuthorized] = "false"
 	}
+	if len(tokenPayload) > 0 {
```
Comment (nkoblai) [inline jwt/service/jwt_token_service.go:180]: Does this if makes sense?

File: user_payload_provider/http/provider.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	sha := hex.EncodeToString(h.Sum(nil))
+
+	client := &http.Client{}
+	request, err := http.NewRequest("POST", p.url, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, fmt.Errorf("creating http client fro http user payload provider: %v", err)
+	}
+	request.Header.Set("Digest", "SHA-256="+sha)
+	request.Header.Set("Content-type", "application/json")
+	resp, err := client.Do(request)
```
Comment (nkoblai) [inline user_payload_provider/http/provider.go:67]: I guess you should use defer resp.Body.Close() after this.

File: user_payload_provider/mock/provider.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package mock
```
Comment (nkoblai) [inline user_payload_provider/mock/provider.go:1]: I'm not sure, but exported methods should have comments. Maybe that can be omitted for mock packages.

File: web/api/login.go [hunk 4]
```diff
@@ -145,10 +153,31 @@ func (ar *Router) IsLoggedIn() http.HandlerFunc {
 	}
 }
 
+// getTokenPayloadForApp get additional token payload data
+func (ar *Router) getTokenPayloadForApp(app model.AppData, user model.User) (map[string]interface{}, error) {
+	if app.TokenPayloadService() == model.TokenPayloadServiceHttp {
+		// check if we have service cached
+		ps, exists := ar.tokenPayloadServices[app.ID()]
+		if !exists {
```
Comment (nkoblai) [inline web/api/login.go:161]: Maybe its better to use early return here?
`if exists { return ps.TokenPayloadForApp(app.ID(), app.Name(), user.ID()) } `

File: jwt/service/jwt_token_service.go (hunk 1 above)
Comment (erudenko) [reply jwt/service/jwt_token_service.go:180]: why not?

**Pull request merged**